aggregators/converter.go

```diff
@@ -45,80 +252,62 @@ func EventToCombinedMetrics(
 	p Partitioner,
 	callback func(CombinedMetricsKey, *aggregationpb.CombinedMetrics) error,
 ) error {
-	collector := combinedMetricsCollector{}
-	svcKey := serviceKey(e, unpartitionedKey.Interval)
-	svcInstanceKey, err := serviceInstanceKey(e)
+	globalLabels, err := marshalEventGlobalLabels(e)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to marshal global labels: %w", err)
 	}
-	hasher := Hasher{}.
-		Chain(svcKey).
-		Chain(svcInstanceKey)
 
-	switch e.Type() {
-	case modelpb.TransactionEventType:
-		repCount := e.GetTransaction().GetRepresentativeCount()
-		if repCount <= 0 {
-			return nil
-		}
-		hdr := hdrhistogram.New()
-		hdr.RecordDuration(e.GetEvent().GetDuration().AsDuration(), repCount)
-
-		txnKey, sim := eventToTxnMetrics(e, hdr)
-		collector.add(p.Partition(hasher.Chain(txnKey)), sim)
-
-		svcTxnKey, sim := eventToServiceTxnMetrics(e, hdr)
-		collector.add(p.Partition(hasher.Chain(svcTxnKey)), sim)
+	pmb := partitionedMetricsBuilderPool.Get().(*partitionedMetricsBuilder)
+	defer func() {
+		// don't use ResetVT, we're not using VT pools here
+		pmb.ServiceInstanceMetrics.Reset()
```

Comment:
Good call, added `getX` functions to get pooled builders, and added `release` methods to them.

Reply:
Actually we don't need to reset any of them, so that's probably making things unclear. I've removed the Reset call now.

Let me know if things are clear enough now with the new release methods, or if further docs would be helpful.

Reply:
LGTM!